### ISSUE TITLE
api issue

### ISSUE THREAD

**Issue by vpapas** [netdata/netdata#6835]
Hello,

After upgrading netdata to the latest version, netdata API doesn't return data for system load for interval less than 5 seconds:

```
QUERY: http://server:19999/api/v1/data?chart=system.load&after=-4&options=seconds
OUTPUT:
{
 "labels": ["time", "load1", "load5", "load15"],
    "data":
 [

  ]
}
```

If i query data for 5 seconds or higher API returns data:
QUERY:http://server:19999/api/v1/data?chart=system.load&after=-5&options=seconds
OUTPUT:
{
 "labels": ["time", "load1", "load5", "load15"],
    "data":
 [
      [ 1568302570, 0, 0.11, 0.11]
  ]
}

This is only for system load. For example, system ram works fine.

```
QUERY: http://server:19999/api/v1/data?chart=system.ram&after=-2&options=seconds
OUTPUT: 
{
 "labels": ["time", "free", "used", "cached", "buffers"],
    "data":
 [
      [ 1568302700, 245.5977, 572.6953, 1001.2617, 0],
      [ 1568302699, 245.7188, 572.5742, 1001.2617, 0]
  ]
}
```

netdata version: v1.17.0-53-g4c9d6a3

**Comment by vpapas** [comment]
Here is the conf:

`# memory mode = save`

**Comment by mfundul** [comment]
This is very possibly due to the query engine refactoring to support variable granularity. In this work, all relative queries have been aligned to database interval granularity and it is impossible to get a result when asking for smaller than database intervals.

What it used to be like before v1.17:
before=ABSOLUTE after=RELATIVE -> before=unaligned after=aligned (this case)
before=RELATIVE after=ABSOLUTE -> before=aligned after=unaligned
before=RELATIVE after=RELATIVE -> before=aligned after=aligned
before=ABSOLUTE after=ABSOLUTE -> before=unaligned after=unaligned

What it is after v1.17:
before=ABSOLUTE after=RELATIVE -> before=aligned after=aligned (this case)
before=RELATIVE after=ABSOLUTE -> before=aligned after=aligned
before=RELATIVE after=RELATIVE -> before=aligned after=aligned
before=ABSOLUTE after=ABSOLUTE -> before=unaligned after=unaligned

I do consider the new behaviour as the original intention due to the original code that was:
```
    long available_points = duration / st->update_every;

    if(duration <= 0 || available_points <= 0)
        return rrdr_create(st, 1);
```

In my opinion the new behaviour is more correct, but we can discuss the implications and maybe take corrective action.

**Comment by ktsaou** [comment]
Let's assume that we have 1 point per day.
The user queries for a point of 10 mins within this day.
Which value we should return?

As @mfundul suggested, the best value is the value of the day.

**Comment by mfundul** [comment]
I will validate this behaviour at a later time, but at the moment I should at least make restore the old behaviour in @vpapas 's example. This way we keep the old behaviour so as to not surprise our users. Making this a high priority.

**Comment by mfundul** [comment]
@vpapas can you please verify that your problem is fixed with #6938?
